### DISCUSSION TITLE
Error while enumerating Datamodule

### DISCUSSION THREAD

**Discussion by pranaycv** [open-edge-platform/anomalib#2195] [Q&A] [answered]
I am loading custom dataset fro folder, thereafter trying to enumerate the dataset. 

But landed into following error.
```bash
RuntimeError: DataLoader worker (pid(s) 15836, 27364, 27808, 30772, 24408, 27804, 112476, 111348) exited unexpectedly
```

Snapshot of the code block is given below:
```python
datamodule = Folder(
    name="vials",
    root="E:\PROJECTS\TestAnomalib\TrainSET",
    normal_dir="good",
    # abnormal_dir="abnormal",
    normal_split_ratio=0.2,
    extensions=".bmp",
    # image_size=(256, 256),
    train_batch_size=8,
    eval_batch_size=4,
    task=TaskType.CLASSIFICATION,
)
datamodule.setup()

i, data = next(enumerate(datamodule.val_dataloader()))
print(data.keys())
```
If anybody could help me on this. it would be of great help.

**Comment by samet-akcay** [comment, marked as answer]
It sounds like workers crashed for some reason. Can you set `num_workers=0` in `Folder` and try again?

**Reply by pranaycv** [reply]
Thanks Samet,
It worked, amazing. 
Hope there are not implications to the performance by setting num_workers=0

**Comment by samet-akcay** [comment]
`num_workers` is related to how many subprocesses you use to load your data. Depending on some factors such as the size, some of the subprocesses might crash, which is probably what happened in your case.

if you set it to 0, which is actually default in pytorch dataloader, the data will be loaded in the main process. For more details you could refer to the following documentation:
https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader
